### PR TITLE
Install Chrony and disable NTP for 1804

### PR DESCRIPTION
**Pull request by adilliadil**
Ntp of ubuntu 1804 has switched to *.ubuntu.pool.ntp.org and is using dynamic ip addresses, this will cause a headache for customer firewall settings.
We need to use chrony as time-sync service on ubuntu 1804
more details:
https://github.com/Azure/AKS/issues/2118

This change disables Ntp & systemd-timesync uses chrony instead

### CODE REVIEW

**Comment by xuto2** [comment]
can you put a description as to why we need to do so?

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -266,6 +266,68 @@ configureCNIIPTables() {
     fi
 }
 
+disableNtpAndTimesyncdInstallChrony() {
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_config.sh:269]: Move to cse_install_ubuntu.sh. We will ask Mariner team to help on the Mariner version.

Comment (xuto2) [reply parts/linux/cloud-init/artifacts/cse_config.sh:269]: FYI @hbeberman

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 2]
```diff
@@ -40,6 +33,15 @@ source {{GetCSEInstallScriptDistroFilepath}}
 wait_for_file 3600 1 {{GetCSEConfigScriptFilepath}} || exit $ERR_FILE_WATCH_TIMEOUT
 source {{GetCSEConfigScriptFilepath}}
 
+UBUNTU_RELEASE=$(lsb_release -r -s)
+if [[ ${UBUNTU_RELEASE} == "16.04" ]]; then
+    sudo apt-get -y autoremove chrony
+    echo $?
+    sudo systemctl restart systemd-timesyncd
+else
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_main.sh:41]: let's be specific here for 18.04 only

**Review by xuto2 — CHANGES_REQUESTED** [review]
a couple comments, otherwise lgtm

File: parts/linux/cloud-init/artifacts/cse_main.sh (hunk 2 above)
Comment (adilliadil) [reply parts/linux/cloud-init/artifacts/cse_main.sh:41]: sure

**Comment by adilliadil** [comment]
> a couple comments, otherwise lgtm. also is it possible to add a (hacky) test in the test vhd part before we have e2e test?

yes discussed offline, created a task for this: [Task 9460707](https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/9460707): Test Chrony in the CSE

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 3]
```diff
@@ -40,6 +40,10 @@ source {{GetCSEInstallScriptDistroFilepath}}
 wait_for_file 3600 1 {{GetCSEConfigScriptFilepath}} || exit $ERR_FILE_WATCH_TIMEOUT
 source {{GetCSEConfigScriptFilepath}}
 
+if [[ ${UBUNTU_RELEASE} == "18.04" ]]; then
+    disableNtpAndTimesyncdInstallChrony
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_main.sh:44]: can we put this behind a toggle (default on)?

**Pull request merged**